src/lib.rs

```diff
@@ -78,14 +78,23 @@ pub trait Load: 'static + Sized {
   /// ```ignore
   ///     Ok(Loaded::with_deps(the_resource, vec![a_key.into(), another_key.into()]))
   /// ```
-  fn load(key: Self::Key, storage: &mut Storage) -> Result<Loaded<Self>, Self::Error>;
+  fn load(
+    key: Self::Key,
+    ctx: &mut C,
+    storage: &mut Storage<C>,
```

Comment:
Ok, `Storage<C>` and `ctx` here. Seems obvious. I guess some helpers could be useful for when people want `C = ()`.